### ISSUE TITLE
Weird error when building the project on different machines

### ISSUE THREAD

**Issue by Riccardo-Andreatta** [getsentry/sentry-electron#143]
Hi there,
we are having issues with our local build on different machines with same version of TypeScript installed (mine and my colleague one). This is the error that we have:
```
ERROR in /Users/<user>/www/<project_name>/node_modules/@sentry/types/dist/index.d.ts (219,15): In ambient enum declarations member initializer must be constant expression.
```
but this is not happening on my local environment (Angular).

We have an Angular front end project and I have created a separate module for managing Sentry only on particular environments (production and staging in this case).

We have also many external plugins/libraries in the project that are causing `Non-Error exception captured` exceptions that we do not want to track and send to Sentry as we are not interested in them. We basically want to track just our platform errors and exceptions.

For this reason, in my Angular module, I have implemented a check in the `beforeSend` like this:

```JavaScript
Sentry.init({
    dsn: 'https://<key>@sentry.io/<project>',
    beforeSend(event, hint) {
        if (
            event.exception.values[0].value.startsWith('Non-Error exception captured')
            || hint.originalException['message'].startsWith('Non-Error exception captured')
        ) {
            // We want to ignore those kind of errors
            return null;
        }
        return event;
    }
})
```

I am still not sure about this check because it is checking `strings` so I was wondering if there is something more smart already implemented and/or if there are some sort of filtering that we may use instead.

P.S.: I may had to post this issue in another repository, sorry about that... 😄

**Comment by timfish** [comment]
You need to be using TypeScript >= 2.4

**Comment by timfish** [comment]
We have just released v3.0.0 of the SDK which includes a lot of changes and improvements since this issue was opened. 

If you can still reproduce this with v3, please open a new issue and I'll look into it.
